### PR TITLE
docs: Add tutorials for Java chat room app

### LINKED ISSUE #1523: Add Java code samples in topics under Develop
My understanding is that this is a PM ask. Looks like this is for the building an app/client libraries context.

Docs to add java code examples to appropriate topics under:
https://docs.redpanda.com/docs/develop/

cc: @mattschumpert  @StasiaZam

### PR DESCRIPTION
**Pull request by JakeSCahill**
Review deadline: May 31

Fixes #1523

 - Added support for Java syntax highlighting using this workaround: https://github.com/facebook/docusaurus/issues/7209.
 - Added a check for XML code samples in the editable placeholders script.

### CODE REVIEW

**Review by Deflaimun — APPROVED** [review]
LGTM from website engine side.

File: docs/develop/guide-java-cloud.mdx [hunk 1]
```diff
@@ -0,0 +1,432 @@
...
+<ContextLink frontmatter={frontMatter}
+/>
+
+Create a basic chat room application with Redpanda Cloud and [Kafka Java clients](https://central.sonatype.com/artifact/org.apache.kafka/kafka-clients).
+
+<Intro/>
+
+## Prerequisites
+
+You must have
```
Comment (Feediver1) [inline docs/develop/guide-java-cloud.mdx:34]:
```suggestion
You must have:
```

**Review by Feediver1 — APPROVED** [review]

File: docs/develop/guide-java.mdx [hunk 2]
```diff
@@ -0,0 +1,413 @@
...
+  Output:
+
+  ```
+  TOPIC      STATUS
+  chat-room  OK
+  ```
+
+## Confirm that the topic exists on the client side
+
+The client may not always know that the topic exists. You can verify that it exists and create it if not. In `chat-room/` create a source file called `src/main/java/example/Admin.java` with the following content.
```
Comment (michael-redpanda) [inline docs/develop/guide-java.mdx:163]: The source paths here are missing `com`:

`src/main/java/com/example/`

This is true in the cloud instructions as well

**Review by michael-redpanda — COMMENTED** [review]
Looks good!  One minor issue with the source file paths but everything built and worked as expected

File: docs/develop/guide-java.mdx (hunk 2 above)
Comment (michael-redpanda) [reply docs/develop/guide-java.mdx:163]: This applies to all of the java source paths (so a find/replace should be a quick fix)

Comment (JakeSCahill) [reply docs/develop/guide-java.mdx:163]: @michael-redpanda It worked for me without including the `com/` path. Is this a convention that we should follow?

Comment (michael-redpanda) [reply docs/develop/guide-java.mdx:163]: `mvn` will create a directory structure based on the group provided (so for instance the instructions say `-DgroupId=com.example`).  So the tree created will be `src/main/java/com/example/`  at least that's what `mvn` did for me.

I think to reduce confusion you should modify the paths.

Comment (michael-redpanda) [reply docs/develop/guide-java.mdx:163]: Except for the paths, I copied everything exactly as it was in the instructions and it worked fine when I placed the files in `src/main/java/com/example` so there's no changes required within the code or `mvn` commands.

**Review by michael-redpanda — APPROVED** [review]
LGTM!

File: docs/develop/guide-java.mdx (hunk 2 above)
Comment (JakeSCahill) [reply docs/develop/guide-java.mdx]: Thanks, updated 😄

**Pull request merged**